utils.go

```diff
@@ -192,3 +192,24 @@ func redactSignature(origAuth string) string {
 	// Strip out 256-bit signature from: Signature=<256-bit signature>
 	return regSign.ReplaceAllString(newAuth, "Signature=**REDACTED**")
 }
+
+// Get default location returns the location based on the input
+// URL `u`, if none of the standard URL matches then if regionOverride
+// is given location gets defaulted to that instead.
+//
+// If no other cases match then the location is set to `us-east-1`
+// as a last resort.
+func getDefaultLocation(u url.URL, regionOverride string) (location string) {
+	// Default to location to 'us-east-1'.
+	switch {
+	case s3utils.IsAmazonChinaEndpoint(u):
+		location = "cn-north-1"
+	case s3utils.IsAmazonGovCloudEndpoint(u):
+		location = "us-gov-west-1"
+	case regionOverride != "":
+		location = regionOverride
+	default:
+		location = "us-east-1"
+	}
+	return
```

Comment:
actually this is more readable as it is not a if/else situation (and hence switch need not be used):
```
        func getDefaultLocation(u url.URL, regionOverride string) (location string) {                                                                                                                              
                if s3utils.IsAmazonChinaEndpoint(u) {                                                                                                                                                              
                        return "cn-north-1"                                                                                                                                                                        
                }                                                                                                                                                                                                  
                if s3utils.IsAmazonGovCloudEndpoint(u) {                                                                                                                                                           
                        return "us-gov-west-1"                                                                                                                                                                     
                }                                                                                                                                                                                                  
                if regionOverride != "" {                                                                                                                                                                          
                        return regionOverride                                                                                                                                                                      
                }                                                                                                                                                                                                  
                return "us-east-1"                                                                                                                                                                                 
        }            
```